src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,253 @@
+package storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.Duke;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+public class Storage {
+    /** Relative file path to the folder containing the data file */
+    private String filePath;
+    /** Name of data file */
+    private String fileName;
+
+    /**
+     * Constructor for Storage class.
+     * @param filePath Relative file path to the folder containing the data file.
+     * @param fileName Name of data file.
+     */
+    public Storage(String filePath, String fileName) {
+        this.filePath = filePath;
+        this.fileName = fileName;
+    }
+
+    /**
+     * Returns ArrayList of Tasks loaded in from a data file.
+     * @return ArrayList of Tasks listed in data file.
+     * @throws IOException If reading file or file directory has issues.
+     */
+    public ArrayList<Task> load() throws IOException {
```

Comment:
Might I suggest extracting out the logic for parsing from the data file to follow the Single Level of Abstraction Principle (SLAP)?